api/python/provisioner/commands/release/release.py

```diff
@@ -0,0 +1,291 @@
+        #   be consistent
+        if self._metadata_url:
+            if self._version:
+                logger.warning(
+                    "'version' parameter would be ignored since"
+                    "'metadata_url' is provided"
+                )
+                self._version = None
+            if self._metadata:
+                logger.warning(
+                    "'metadata' parameter would be ignored since"
+                    "'metadata_url' is provided"
+                )
+                self._metadata = None
+        elif self._metadata:
+            if self._version and self._version != self._metadata.version:
+                raise ValueError(
+                    "'version' parameter '{self._version}' doesn't match "
+                    "metadata info: '{self._metadata.version}'"
+                )
+        elif not self._version:
+            raise ValueError(
+                "either 'version' or 'metadata' or 'metadata_url' "
+                "should be defined"
+            )
+
+    @property
+    def is_factory(self):
+        factory_release = self.factory_release()
+        return (
+            self is factory_release
+            or self.version == factory_release.version
+        )
+
+    @property
+    def iso_version(self) -> config.ISOVersion:
+        """
+        Return version of ISO bundle scheme for the release
+
+        Returns
+        -------
+        config.ISOVersion:
+            return version of ISO bundle scheme for the release
+
+        """
+        if self._iso_version is None:
+            if self.is_factory:
+                # FIXME improve to support newer versions for deployment
+                self._iso_version = config.ISOVersion.VERSION1
+            else:
+                # search in upgrade releases
+                pillar_path = f'release/upgrade/repos/{self.version}'
+                pillars = PillarResolver(local_minion_id()).get(
+                    [PillarKey(pillar_path)],
+                    fail_on_undefined=True
+                )
+
+                upgrade_release = pillars[local_minion_id()][
+                    PillarKey(pillar_path)
+                ]
+                release_data = upgrade_release.get(self.version)
+                # NOTE: release_data can be None after applying
+                #  remove_swupgrade_repo command
+                if (
+                    isinstance(release_data, dict)
+                    and 'version' in release_data
+                ):
+                    self._iso_version = config.ISOVersion(
+                        release_data['version']
+                    )
+                else:
+                    # FIXME: it may be remote repo
+                    self._iso_version = config.ISOVersion.VERSION1
+
+        return self._iso_version
+
+    @property
+    def metadata_url(self) -> CortxReleaseUrl:
+        """
+        Return URL to a Cortx repository metadata
```

Comment:
Just a suggestion. Maybe it is better to name it 'URI', not the 'URL'. Because URL is a `http`/`https`/`ftp` scheme, but `file://` is a file URI scheme. So users can expect that this method returns the path to the remote resource instead of the path to the local file. URI is more a broader concept than URL.

Reply:
correct, I updated that. Thanks